internal/rbd/group.go

```diff
@@ -71,12 +71,34 @@ func (rv *rbdVolume) RemoveFromGroup(ctx context.Context, vg types.VolumeGroup)
 		return fmt.Errorf("could not get iocontext for volume group %q: %w", vg, err)
 	}
 
-	name, err := vg.GetName(ctx)
+	id, err := vg.GetID(ctx)
 	if err != nil {
-		return fmt.Errorf("could not get name for volume group %q: %w", vg, err)
+		return fmt.Errorf("could not get id for volume group %q: %w", vg, err)
+	}
+
+	return librbd.GroupImageRemove(ioctx, id, rv.ioctx, rv.RbdImageName)
```

Comment:
`GroupImageRemove` accepts the group name and not id, so shouldn't we continue to fetch the name and pass it to the API? Or, am I missing something here?

Reply:
The naming has been a bit confusing to me as well. The `GetID()` function returns the generated unique name of the rbd-group (something like `csi-group-<UUID>`), the `GetName()` function returns the name that was requested by the caller with the `CreateVolumeGroup` procedure (stored in the journal and pointing to the generated name/id of the real RBD-group).

Throughout Ceph-CSI there are many `id` fields, and the naming has not always been consistent. This commit corrects an earlier mistake that I found when manually creating a volume-group with a name `my-group` and trying to use that to create a VolumeGroupSnapshot of. ID/name was incorrect, with this change it looks way cleaner.

Reply:
I'll add a better description in `internal/rbd/types.journalledObject` for these functions.